aws-iot-certificate/src/main/java/com/amazonaws/iot/certificate/DeleteHandler.java

```diff
@@ -74,16 +74,14 @@ public DeleteHandler(IotClient iotClient) {
             proxy.injectCredentialsAndInvokeV2(deleteRequest, iotClient::deleteCertificate);
             logger.log(String.format("%s [%s] deleted successfully", ResourceModel.TYPE_NAME, certificateId));
 
-        } catch (final CertificateStateException e) {
-            throw new CfnGeneralServiceException(ResourceModel.TYPE_NAME, e);
+        } catch (final CertificateStateException | ServiceUnavailableException e) {
+            throw new CfnGeneralServiceException(operation, e);
```

Comment:
>  CertificateStateException
>     The certificate operation is not allowed.
>     HTTP Status Code: 406

Doesn't seem like it should be translated to a CfnGeneralServiceException

Reply:
Translating this to `CfnInvalidRequestException`.